src/data/nat/prime.lean

```diff
@@ -863,6 +863,20 @@ begin
       nat.prod_factors hb]
 end
 
+lemma pow_factors_count_dvd {n p : ℕ} (hp : p ≠ 0) :
+  p ^ n.factors.count p ∣ n :=
+begin
```

Comment:
```suggestion
lemma pow_factors_count_dvd {n p : ℕ} : p ^ n.factors.count p ∣ n :=
begin
  rcases eq_or_ne p 0 with rfl | hp,
  { convert one_dvd _,
    convert pow_zero _,
    rw count_eq_zero_of_not_mem,
    exact mt prime_of_mem_factors not_prime_zero },
```

idk if you'll want to change explicitness of stuff after this

Reply:
(or golf it this proof is atrocious)

Reply:
I am not sure if you like this proof better, but this also works:
```suggestion
lemma pow_factors_count_dvd {n p : ℕ} : p ^ n.factors.count p ∣ n :=
begin
  rcases eq_or_ne p 0 with rfl | hp,
  { rw count_eq_zero_of_not_mem (mt prime_of_mem_factors not_prime_zero),
    simp },
```

Reply:
I found a proof shorter than all three of ours! Thanks for the suggestion :)